src/block/block_compact.c

```diff
@@ -205,9 +205,23 @@ __block_compact_skip_internal(WT_SESSION_IMPL *session, WT_BLOCK *block, bool es
     __wt_verbose_level(session, WT_VERB_COMPACT,
       (estimate ? WT_VERBOSE_DEBUG_3 : WT_VERBOSE_DEBUG_1),
       "%s:%s require 10%% or %" PRIuMAX "MB (%" PRIuMAX
-      ") in the first 90%% of the file to perform compaction, compaction %s",
+      ") in the first 90%% of the file to perform compaction",
       block->name, estimate ? " estimating --" : "", (uintmax_t)(file_size / 10) / WT_MEGABYTE,
-      (uintmax_t)(file_size / 10), *skipp ? "skipped" : "proceeding");
+      (uintmax_t)(file_size / 10));
+
+    /* Skip files that have failed to make progress on previous compact iterations. */
+    if (block->compact_estimated && !*skipp) {
```

Comment:
Why are we checking for `block->compact_estimated`? Is this a way to check whether there was a previous iteration? If so, you might want to explain it in a comment.

Reply:
Since the estimation phase is not supposed to rewrite blocks, we should make sure we are doing this test when we are proceeding with compaction. The `block->compact_estimated` tells us we are after the estimation phase. I agree that we could have a comment explaining this.

Reply:
added to the comment in https://github.com/wiredtiger/wiredtiger/pull/10447/commits/41055f53ac889416a650116dc00a6c12b749744b